### ISSUE TITLE
DJ List Adjustments

### ISSUE THREAD

**Issue by lebull** [SundayServiceVR/sunday-service-vr#94]
Talking with our first real user, there's a desire to see more of the dj list in the stream.  

```
### Easy Option
Bootstrap grid / margin / inline form labels to squeeze it in.

### Another Option
Split slot information into tabs.

https://cdn.discordapp.com/attachments/1223694818144813228/1264261913018237000/image.png?ex=669d3b28&is=669be9a8&hm=3e161720263df6704c0f5efd96f0331d610be97b6834466b8988d3e457a0c9d5&
```

**Comment by lebull** [comment]
My thought was to use a model with a form to adjust set information.

**Comment by atomicpeach** [comment]
for creating a schedule, we really shouldn't need to have multiple tabs or modals. What was the root issue with the existing format?

**Comment by lebull** [comment]
So I don't suspect we'll need tabs, but I do suspect we'll want to use a modal to adjust slot details.

We're going to be adding more stuff to slots:  The ability to change the slot name, marking a slot as an afterparty, better support for b2b's, and it's putting a lot of info on the screen per slot.  I think we should let the current lineup editor focus on adding dj's and the ordering, and we need a separate space to adjust the details once the lineup is solidified.

The main feedback we got here was they wanted to see more on the screen at a time.  I think what we should focus on is just making sure we're showing/editing useful information and hiding information that's not useful for any given task if that makes sense.

If ya have any other ideas, lets hear em!

**Comment by atomicpeach** [comment]
I think I'm failing to see how that information _wouldn't_ be wanted when creating the lineup and thus why it should be hidden? I see two decent methods for this, though: create a separate workflow/tab for just the line up generation and then a second one with all of this additional info; or make the current cards collapsible with only the DJ name, slot length, and start time visible in the collapsed form. If we go the latter route (which I would prefer as a host), have buttons on the side to quickly (un)collapse all cards.

Most of the info proposed can easily fit in the existing space if we cleanup the card and make the form fields smaller and/or more inline. The set type doesn't really need to be full width at all (probably could even get away with 50% being buttons and 50% being the link field). The debutt toggle can be 20%. Since the slot length is almost always 30 mins or 60 mins, we can probably swap that to a smol text field with - and + buttons that change in 15 minute and/or 10 minute increments (for a 20 min set) but still allow for text entry if its weird or if that's how the host prefers to enter data.

The alternative names will be used so little that it likely shouldn't be shown for the vast majority of our DJs. We can probably get away with a simply "aka" string like how Discord handles it in DMs.

**Comment by atomicpeach** [comment]
Here's a mock up of an uncollapsed and collapsed card. Graphics aren't my strongest and this is super rough, but I tossed in some extra info regarding timeslot availability, too! (I think we can work on a better display for this, tbh; but that's a later upgrade.) The DJ info wouldn't appear unless the slot name is edited, the DJ info would be what's in the slot name field by default. Maybe we need to add a + button here add B2B DJs? The debutt slider is red/green depending on which side it's on.

This card works on mobile well, too, with everything just collapsing into a vertical column instead of inline fields. Maybe add a "hide availability info" button for mobile displays to keep it a little cleaner assuming the host is trying to reference this info during the event?

![image](https://github.com/user-attachments/assets/e480a415-ceea-461b-8dcd-22e53e42819c)

**Comment by lebull** [comment]
Yeah, lets give your mockup a shot ^^.  If we need to make adjustments beyond that, we will have a better idea of where to go when it's actually implemented.

**Comment by lebull** [comment]
I'm making the executive decision to move forward by splitting up the stream information from signup and lineup information.
